### DISCUSSION TITLE
I can't get ionic cordova app to build for android no matter what i do

### DISCUSSION THREAD

**Discussion by glenelkinsdev** [apache/cordova#385] [Q&A]
So i have an ionic app that uses cordova and it works fine building for ios but i'm having a never ending nightmare with android, i have literally updated all packages in my ionic package.json and i am using cordova-android@10.1.2 - i currently can't use 11 as it just causes even more issues.

Gradle version on my computer is as follows:

------------------------------------------------------------
Gradle 8.0.1
------------------------------------------------------------

Build time:   2023-02-17 20:09:48 UTC
Revision:     68959bf76cef4d28c678f2e2085ee84e8647b77a

Kotlin:       1.8.10
Groovy:       3.0.13
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          1.8.0_333 (Oracle Corporation 25.333-b02)
OS:           Mac OS X 12.6 x86_64

I am targeting SDK version 30 as it won't build with anything higher with the current setup, but i now get these errors:

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:checkReleaseAarMetadata'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
   > The minCompileSdk (31) specified in a
     dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
     is greater than this module's compileSdkVersion (android-30).
     Dependency: androidx.paging:paging-runtime-ktx:3.1.0.
     AAR metadata file: /Users/glenelkins/.gradle/caches/transforms-3/ab5210161278a57dd3ce15e3d01de5e0/transformed/jetified-paging-runtime-ktx-3.1.0/META-INF/com/android/build/gradle/aar-metadata.properties.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:bundleReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.Aapt2ProcessResourcesRunnable
   > Android resource linking failed
     ERROR:/Users/glenelkins/.gradle/caches/transforms-3/d1025da3f26a9398631572d33376f02b/transformed/core-1.7.0/res/values/values.xml:105:5-114:25: AAPT: error: resource android:attr/lStar not found.

I am literally tearing my hair out with this, it builds no problem for iOS and i don't even know what these errors mean, taking into account i'm not a massive app developer i'm more of a backend server developer so i'm a bit stuck on how to get this to build.

Is this telling me that the current gradle version is expecting to build for sdk 31 but i'm targeting 30? So should i be downgrading gradle? Baring in mind right now i cannot upgrade to cordova-android@11 as it causes many more issues and i just need this to build

**Comment by jcesarmobile** [comment]
Update to cordova-android 11 even if there are more errors to fix, forcing a targetSDK in an older cordova-android version that didn't officially support it will just give you problems that are not solvable as most of them will require code changes.
